pkg/statistics/handle/util/util.go

```diff
@@ -100,80 +100,126 @@ func CallWithSCtx(pool syssession.Pool, f func(sctx sessionctx.Context) error, f
 	if err != nil {
-		return err
+		return restore, err
 	}
 	ver, err := strconv.ParseInt(verInString, 10, 64)
 	if err != nil {
-		return err
+		return restore, err
 	}
-	sctx.GetSessionVars().AnalyzeVersion = int(ver)
+	sv.AnalyzeVersion = int(ver)
 
 	// enable historical stats
-	val, err := sctx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBEnableHistoricalStats)
+	val, err := sv.GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBEnableHistoricalStats)
 	if err != nil {
-		return err
+		return restore, err
 	}
-	sctx.GetSessionVars().EnableHistoricalStats = variable.TiDBOptOn(val)
+	sv.EnableHistoricalStats = variable.TiDBOptOn(val)
 
 	// partition mode
-	pruneMode, err := sctx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBPartitionPruneMode)
+	pruneMode, err := sv.GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBPartitionPruneMode)
 	if err != nil {
-		return err
+		return restore, err
 	}
-	sctx.GetSessionVars().PartitionPruneMode.Store(pruneMode)
+	sv.PartitionPruneMode.Store(pruneMode)
 
 	// enable analyze snapshot
-	analyzeSnapshot, err := sctx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBEnableAnalyzeSnapshot)
+	analyzeSnapshot, err := sv.GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBEnableAnalyzeSnapshot)
 	if err != nil {
-		return err
+		return restore, err
 	}
-	sctx.GetSessionVars().EnableAnalyzeSnapshot = variable.TiDBOptOn(analyzeSnapshot)
+	sv.EnableAnalyzeSnapshot = variable.TiDBOptOn(analyzeSnapshot)
 
 	// enable skip column types
-	val, err = sctx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBAnalyzeSkipColumnTypes)
+	val, err = sv.GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBAnalyzeSkipColumnTypes)
 	if err != nil {
-		return err
+		return restore, err
 	}
-	sctx.GetSessionVars().AnalyzeSkipColumnTypes = variable.ParseAnalyzeSkipColumnTypes(val)
+	sv.AnalyzeSkipColumnTypes = variable.ParseAnalyzeSkipColumnTypes(val)
 
 	// skip missing partition stats
-	val, err = sctx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBSkipMissingPartitionStats)
+	val, err = sv.GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBSkipMissingPartitionStats)
 	if err != nil {
-		return err
+		return restore, err
 	}
-	sctx.GetSessionVars().SkipMissingPartitionStats = variable.TiDBOptOn(val)
-	verInString, err = sctx.GetSessionVars().GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBMergePartitionStatsConcurrency)
+	sv.SkipMissingPartitionStats = variable.TiDBOptOn(val)
+	verInString, err = sv.GlobalVarsAccessor.GetGlobalSysVar(vardef.TiDBMergePartitionStatsConcurrency)
 	if err != nil {
-		return err
+		return restore, err
 	}
 	ver, err = strconv.ParseInt(verInString, 10, 64)
 	if err != nil {
-		return err
+		return restore, err
 	}
-	sctx.GetSessionVars().AnalyzePartitionMergeConcurrency = int(ver)
-	return nil
+	sv.AnalyzePartitionMergeConcurrency = int(ver)
+
+	if sv.TimeZone == nil {
+		sv.TimeZone = time.UTC
+	}
+
+	return restore, nil
```

Comment:
restore mechanism looks good to me, since we don't want to pollute the ddl session default variables, while stats module current require stats-approvers @0xPoe